example/cra-provider/src/Web3Buttons.tsx

```diff
@@ -1,22 +1,46 @@
 import { useWeb3React } from '@web3-react/core'
-import { Connector } from '@web3-react/types'
-import { useEffect, useState } from 'react'
+import { useEffect, useRef, useState } from 'react'
 import { useArcxAnalytics } from '@arcxmoney/analytics'
 import { metamask, walletConnect } from './connectors'
 
 export const Web3Buttons = () => {
   const sdk = useArcxAnalytics()
   const { account, connector, chainId, provider } = useWeb3React()
+  const knownChainId = useRef(chainId)
+  const knownAccount = useRef(account)
   const [connectedWallet, setConnectedWallet] = useState<'metamask' | 'walletConnect' | undefined>()
 
+  useEffect(() => {
+    if (!sdk) return
+
+    if (knownAccount.current !== account && account && chainId) {
+      if (account) {
+        // Connect
+        sdk.wallet({
+          chainId,
+          account,
+        })
+      } else {
+        // Disconnect
+        sdk.wallet({
+          chainId,
+          account: '',
+        })
+      }
+      knownAccount.current = account
+    } else if (knownChainId.current !== chainId && chainId) {
+      sdk.chain({ chainId })
+    }
```

Comment:
Should this be an `else`? i.e. is it possible to have a new account AND and new chain in a single useEffect() refresh? 
```suggestion
    } 
    
    if (knownChainId.current !== chainId && chainId) {
      sdk.chain({ chainId })
    }
```

Reply:
Just noticed this is in example code, not really that big a deal ^^ 😅

Reply:
> is it possible to have a new account AND and new chain in a single useEffect() refresh?
No it shouldn't but I tested out this code and it worked well enough, so I'd say it's good enough for the example :)

Reply:
If we do it this way, there will be a CONNECT followed by a CHAIN_CHANGED immediately, so I would leave it as is, esp given this is an example repo